rabbitmq/src/main/java/io/micronaut/rabbitmq/intercept/RabbitMQConsumerAdvice.java

```diff
@@ -75,26 +79,30 @@
     private final RabbitMessageSerDesRegistry serDesRegistry;
     private final ConversionService conversionService;
     private final Map<Channel, ConsumerState> consumerChannels = new ConcurrentHashMap<>();
+    private final RabbitConnectionFactoryConfig connectionFactoryConfig;
 
     /**
      * Default constructor.
      *
-     * @param beanContext       The bean context
-     * @param binderRegistry    The registry to bind arguments to the method
-     * @param exceptionHandler  The exception handler to use if the consumer isn't a handler
-     * @param serDesRegistry    The serialization/deserialization registry
-     * @param conversionService The service to convert consume argument values
+     * @param beanContext             The bean context
+     * @param binderRegistry          The registry to bind arguments to the method
+     * @param exceptionHandler        The exception handler to use if the consumer isn't a handler
+     * @param serDesRegistry          The serialization/deserialization registry
+     * @param conversionService       The service to convert consume argument values
+     * @param connectionFactoryConfig The configuration of the rabbit connection
      */
     public RabbitMQConsumerAdvice(BeanContext beanContext,
                                   RabbitBinderRegistry binderRegistry,
                                   RabbitListenerExceptionHandler exceptionHandler,
                                   RabbitMessageSerDesRegistry serDesRegistry,
-                                  ConversionService conversionService) {
+                                  ConversionService conversionService,
+                                  RabbitConnectionFactoryConfig connectionFactoryConfig) {
```

Comment:
This is a public class not marked with `@Internal` so this is a breaking change. A new constructor can be created and marked with `@Inject` and you'll have to deal with the parameter potentially being null